sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ImplementationBridgeHelpers.java

```diff
@@ -106,13 +109,16 @@ private PartitionKeyHelper() {}
 
         public static void setPartitionKeyAccessor(final PartitionKeyAccessor newAccessor) {
             if (!accessor.compareAndSet(null, newAccessor)) {
-                logger.warn("PartitionKeyAccessor already initialized!");
+                logger.debug("PartitionKeyAccessor already initialized!");
+            } else {
+                logger.info("Setting PartitionKeyAccessor...");
```

Comment:
Why is this info? Should we do debug for this as well?

Reply:
This will be logged exactly once - by the "winning" thread that actually set the accessor AtomicReference. IMO having it is info makes sense because it can be used to tell whether initialization has happened or not from logs.